File: Listas/Lista 12/G.py
n, d = map(int, input().split())

numMinMovimentos = 0
nAnterior = 0

elementos = list(map(int, input().split()))

for i in range(n):
    nAtual = elementos[i]
    
    if nAnterior >= nAtual:
        passosParaAdd = (nAnterior - nAtual + 1) // d
        passosParaAdd += 0 if (nAnterior - nAtual + 1) % d == 0 else 1

        nAtual = nAtual + passosParaAdd * d
        numMinMovimentos += passosParaAdd
    
    nAnterior = nAtual

print(numMinMovimentos)

File: Listas/Lista 12/G.c
#include <stdio.h>

int main()
{
    int n, i, nAtual, nAnterior = 0;
    int d, passosParaAdd, numMinMovimentos = 0;
    scanf("%d %d",&n, &d);

    for(i = 1; i <= n; i++)
    {
        scanf("%d",&nAtual);
        if(nAnterior >= nAtual)
        {
            passosParaAdd = (nAnterior - nAtual + 1)/d;
            passosParaAdd += (nAnterior - nAtual + 1)%d == 0 ? 0 : 1;

            nAtual = nAtual + passosParaAdd*d;
            numMinMovimentos += passosParaAdd;
        }
        nAnterior = nAtual;
    }

    printf("%d\n",numMinMovimentos);
    return 0;
}